dask/dataframe/io/parquet/arrow.py

```diff
@@ -412,6 +413,46 @@ def _read_table_from_path(
                 )
 
 
+def _get_rg_statistics(row_group, col_indices):
+    """Custom version of pyarrow's RowGroupInfo.statistics method
+    (https://github.com/apache/arrow/blob/master/python/pyarrow/_dataset.pyx)
+
+    We use col_indices to specify the specific subset of columns
+    that we need statistics for.  This is more optimal than the
+    upstream `RowGroupInfo.statistics` method, which will return
+    statistics for all columns.
+    """
+
+    if subset_stats_supported:
+
+        def name_stats(i):
+            col = row_group.metadata.column(i)
+
+            stats = col.statistics
+            if stats is None or not stats.has_min_max:
+                return None, None
+
+            name = col.path_in_schema
+            field_index = row_group.schema.get_field_index(name)
+            if field_index < 0:
+                return None, None
+
+            typ = row_group.schema.field(field_index).type
+            return col.path_in_schema, {
+                "min": pa.scalar(stats.min, type=typ).as_py(),
+                "max": pa.scalar(stats.max, type=typ).as_py(),
```

Comment:
@jorisvandenbossche - I'm noticing that quite a bit of time is spent in these `pa.scalar().as_py()` conversions, and it doesn't seem to be necessary in the cases I have been testing (the type/result is the same if I just use `stats.min` and `stats.max`).

Is there a rule of thumb we can use to avoid this conversion whenever possible?  Note that, without this conversion, the `"pyarrow-dataset"` option becomes faster than the legacy option with or without the `_metadata` file.

Also - I'd be happy to get your thoughts on this PR in general :)

Reply:
Yes, I don't think that should be necessary (I need to look back why we are actually doing this on the pyarrow side). 
It might be worth looking at the legacy engine `_process_metadata` method, which basically does the same (getting statistics out of the parquet Metadata objects). 

But so one of the reasons we added the actual parquet `FileMetadata`/`RowGroupMetadata` to the fragment/row group is exactly to enable what you are doing here, as those objects already have this lazy access (in contrast to the dataset specific `RowGroupInfo.statistics` we already had in pyarrow 1.0 and 2.0.

Reply:
>It might be worth looking at the legacy engine _process_metadata method, which basically does the same (getting statistics out of the parquet Metadata objects).

Good point - It seems likely that we can avoid this conversion.

> But so one of the reasons we added the actual parquet FileMetadata/RowGroupMetadata to the fragment/row group is exactly to enable what you are doing here, as those objects already have this lazy access (in contrast to the dataset specific RowGroupInfo.statistics we already had in pyarrow 1.0 and 2.0.

Ah - That makes a lot of sense.  I like having access to the FileMetadata/RowGroupMetadata like this :)